scheduler/replica_scheduler.go

```diff
@@ -45,13 +45,31 @@ func NewReplicaScheduler(ds *datastore.DataStore) *ReplicaScheduler {
 	return rcScheduler
 }
 
+// isFailedReplicaUnused will return true if replica was not used before and marked as failed
+func isFailedReplicaUnused(replica *longhorn.Replica, replicas map[string]*longhorn.Replica) bool {
+	isFirstAttachment := true
+	for _, r := range replicas {
+		if r.Spec.NodeID != "" {
+			isFirstAttachment = false
+			break
+		}
+	}
+	return !isFirstAttachment && replica.Spec.HealthyAt == "" && replica.Spec.FailedAt != ""
```

Comment:
How about check `replica.Spec.HealthyAt` and `replica.Spec.FailedAt` first? So we only go through the replicas when necessary, for ex:

```
if replica.Spec.HealthyAt != "" {
  return false
}

if replica.Spec.FailedAt == "" {
  return false
}

for _, r := range replicas {
  ...
}
```

Reply:
Make it more simple, just to check the replica `HealtyhAt` and `FailedAt`.